src/serializers/graph_extensions.c

```diff
@@ -51,6 +51,13 @@ void Serializer_Graph_SetNode
 		RG_Matrix  M  =  Graph_GetLabelMatrix(g, label);
 		GrB_Matrix m  =  RG_MATRIX_M(M);
 		info = GrB_Matrix_setElement_BOOL(m, true, id, id);
+		if(info == GrB_INVALID_INDEX) {
+			DataBlock_Ensure(g->nodes, id);
+			GrB_Index dim = Graph_RequiredMatrixDim(g);
+			RedisModule_Log(NULL, "notice", "RESIZE LABEL MATRIX %lld", dim);
+			RG_Matrix_resize(M, dim, dim);
+			info = GrB_Matrix_setElement_BOOL(m, true, id, id);
+		}
```

Comment:
Can we get into a situation where this operation can fail?
If so this is a bit too much, I think the minimum required would be:
`RG_Matrix_resize(M, id, id);`